pkg/workload/workload.go

```diff
@@ -155,3 +158,64 @@ func max(v1, v2 int64) int64 {
 	}
 	return v2
 }
+
+// getWorkloadCondition extracts the provided condition from the given status and returns that.
+// Returns nil and -1 if the condition is not present, and the index of the located condition.
+func GetWorkloadCondition(status *kueue.QueuedWorkloadStatus, conditionType kueue.QueuedWorkloadConditionType) (int, *kueue.QueuedWorkloadCondition) {
```

Comment:
returning `*kueue.QueuedWorkloadCondition` sounds redundant, the caller can use the index to look it up.